assets/js/src/segments/dynamic/dynamic_segments_filters/woocommerce.tsx

```diff
@@ -70,21 +85,67 @@ export const WooCommerceFields: React.FunctionComponent<Props> = ({ onChange, it
         </div>
       </div>
     );
-  }
+  } else if (item.action === WooCommerceActionTypes.PURCHASED_CATEGORY) {
+    optionFields = (
+      <div className="mailpoet-form-field">
+        <div className="mailpoet-form-input mailpoet-form-select">
+          <Select
+            placeholder={MailPoet.I18n.t('selectWooPurchasedCategory')}
+            options={categoryOptions}
+            value={find(['value', item.category_id], categoryOptions)}
+            onChange={(option: SelectOption): void => compose([
+              onChange,
+              assign(item),
+            ])({ category_id: option.value })}
+          />
+        </div>
+      </div>
+    );
+  } else if (item.action === WooCommerceActionTypes.NUMBER_OF_ORDERS) {
+    if (!item.number_of_orders_type) {
+      item.number_of_orders_type = '=';
+    }
 
-  return (
-    <div className="mailpoet-form-field">
-      <div className="mailpoet-form-input mailpoet-form-select">
-        <Select
-          placeholder={MailPoet.I18n.t('selectWooPurchasedCategory')}
-          options={categoryOptions}
-          value={find(['value', item.category_id], categoryOptions)}
-          onChange={(option: SelectOption): void => compose([
-            onChange,
-            assign(item),
-          ])({ category_id: option.value })}
-        />
+    optionFields = (
+      <div>
+        <Grid.CenteredRow className="mailpoet-form-field">
+          <Select
+            options={numberOfOrdersTypeOptions}
+            value={find(['value', item.number_of_orders_type], numberOfOrdersTypeOptions)}
+            onChange={(option: SelectOption): void => compose([
```

Comment:
I copied this onChange from the other action types but I don't fully understand what is happening here.

Reply:
> I copied this onChange from the other action types
I understand. This code is a result of a couple of iterations and could be probably simplified now.
Let me explain: 

- `compose` gets a list of functions and it calls them from the end and passes the result of the previous call to the next call.
- `assign` takes two objects and assigns the properties from the second one into the first one. 
- both functions are curried so they are called only after all parameters are passed.

The equivalent code is: `onChange(assign(item, { number_of_orders_type: option.value }))`. That can be simplified even more using only JS vanilla.

Reply:
Thanks for the explanation!